### PR TITLE
refactor: Create new compose-builder for generating compose files 

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/developer-scripts/blob/master/.github/Contributing.md.

## What is the current behavior?

source files for generated compose files all under releases/nightly-build/compose-files/source

## What is the new behavior?

source files for generated compose files now under top level compose-generator folder.

No change in functionality.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

no

## Other information

### CODE REVIEW

File: compose-builder/Makefile [hunk 1]
```diff
@@ -86,15 +86,20 @@ ifeq (true, $(DS_GROVE))
...
+ifneq (nexus, $(RELEASE))
+	COMPOSE_FOLDER:=../releases/$(RELEASE)/compose-files
+endif
+
 help:
 	echo "See README.md in this folder"
 
 portainer:
-	cd ..; \
+	cd ./releases/nightly-build/compose-files; \
```
Comment (bnevis-i) [inline compose-builder/Makefile:98]: Have you considered

```
make -C releases/nightly-build/compose-files portainer
```

instead

File: releases/nightly-build/compose-files/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This folder contains the docker compose files that pull and run the EdgeX images
```
Comment (andresrinivasan) [inline releases/nightly-build/compose-files/README.md:4]: I found the multiple README.md files confusing. I would prefer the compose builder README.md documents everything and remove this one.

File: releases/nightly-build/compose-files/Makefile [hunk 3]
```diff
@@ -41,7 +41,7 @@ portainer-down:
 	docker-compose -p portainer -f docker-compose-portainer.yml down
 
 build:
```
Comment (andresrinivasan) [inline releases/nightly-build/compose-files/Makefile:43]: Can we move make build up to compose builder and eliminate the target here?

**Comment by andresrinivasan** [comment]
This iteration/consolidation is a welcome improvement and would have simplified my developer experience for Hanoi. What's still a little confusing is how to be a developer and use locally created containers. I figured it out. Others will figure it out. But it does take a bit of sleuthing.

I suggest the next step (post Hanoi) is to break out the compose-builder and Docker compose component files into their own repo such that it creates a Docker image that can be used directly by edgex-go. This means that with only the edgex-go repo, I can create and use any of the compose files by running the container. This also means we don't need to keep the generated files in the repo and we control the build environment to mitigate challenges like unexpected behavior from docker-compose when it is used to create the compose files.

File: releases/nightly-build/compose-files/README.md (hunk 2 above)
Comment (lenny-intel) [reply releases/nightly-build/compose-files/README.md:4]: The READMEs are not the same. 
This one is targeted to users. 
The one in compose-builder is targeted to developers and those users creating custom compose files.

File: releases/nightly-build/compose-files/Makefile (hunk 3 above)
Comment (lenny-intel) [reply releases/nightly-build/compose-files/Makefile:43]: Yes, It is currently in both makefiles and now with the move, I agree it is no longer needed at this "user" level.

**Comment by lenny-intel** [comment]
@andresrinivasan ,

> What's still a little confusing is how to be a developer and use locally created containers.

We have plans to add the docs on how to use the compose-builder to create custom compose files, which can include the DEV version of the compose file that uses your locally built edgex-go docker images. Would that help? or a separate README-How-to-use-Dev-Images.md in the compose-builder?


> This also means we don't need to keep the generated files in the repo

We specifically build and commit the standard compose files so that user and TestQA can simply pull the appropriate file and run edgeX. The current docs point to these commit compose files in the getting started guide.

File: compose-builder/README.md [hunk 4]
```diff
@@ -51,7 +56,7 @@ This folder contains the following compose files:
 This folder contains the following environment files:
 
 - **.env**
-    This file contains the registry and image version variables referenced in compose files. Docker compose implicitly uses the ".env" file, if it exists, so you will not see it referenced in the compose files. It is referenced in the Makefile so that it can also use these settings.
+    This file contains the the `version`, `repositories` and image `version` variables referenced in compose files. Docker compose implicitly uses the `.env` file, if it exists, so you will not see it referenced in the compose files. It is referenced in the Makefile so that it can also use these settings.
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply compose-builder/README.md:59]: remove the extra `the`

File: compose-builder/README.md [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 > **Note to Developers**: 
-> *Once you have edited and tested your changes to these source files you **MUST** regenerate the composes using the `Makefile` one level up. See `build-all` in the accompanying [README](../README.md) for details.*
+> *Once you have edited and tested your changes to these source files you **MUST** regenerate the composes using the `make build` command.*
+
+Do the following to build compose files for next release such as `hanoi` 
+
+1. Update the `RELEASE`, `REPOSITORY`, `CORE_EDGEX_REPOSITORY` and `versions` contained in the `.env` file.
+2. Create the release folder, i.e `release/hanoi/compose-files`
+3. Run `make build` 
+4. **Undo changes made to `.env` file**. (committed values **must** remain as `nexus/nightly-build`)
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply compose-builder/README.md:15]: Yea, I think we'll end up committing a version of the file like `.env-hanoi` but plan on figuring that out when we do the Hanoi release.

File: compose-builder/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (lenny-intel) [reply compose-builder/README.md]: Fixed

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by bill-mahoney — APPROVED** [review]
LGTM

**Review by jpwhitemn — APPROVED** [review]
LGTM for this release.  In the future, couldn't the make file create the release/[name]/compose-files directory from the .env?  Also, perhaps we need a cleanup script to "undo changes to .env file" before the PR is opened (less reliance on human error).  Again, these are all changes that could be made going forward.

It all worked for me when testing creation of making a Hanoi set of compose files.

**Pull request merged**